src/Spark.Engine.Test/Service/PatchApplicationServiceTests.cs

```diff
@@ -0,0 +1,384 @@
+﻿namespace Spark.Engine.Test.Service
+{
+    using System.Linq;
+    using System.Reflection;
+    using Engine.Extensions;
+    using Engine.Service.FhirServiceExtensions;
+    using Hl7.Fhir.Model;
+    using Hl7.Fhir.Serialization;
+    using Xunit;
+    using Task = System.Threading.Tasks.Task;
+
+    public class PatchApplicationServiceTests
+    {
+        [Fact]
+        public void CanApplyPropertyAssignmentPatch()
+        {
+            var xml = @"<Parameters xmlns=""http://hl7.org/fhir"">
+  <parameter>
+    <name value=""operation""/>
+    <part>
+      <name value=""type""/>
+      <valueCode value=""add""/>
+    </part>
+    <part>
+      <name value=""path""/>
+      <valueString value=""Patient""/>
+    </part>
+    <part>
+      <name value=""name""/>
+      <valueString value=""birthDate""/>
+    </part>
+    <part>
+      <name value=""value""/>
+      <valueDate value=""1930-01-01""/>
+    </part>
+  </parameter>
+</Parameters>";
+            var parser = new FhirXmlParser();
+            var parameters = parser.Parse<Parameters>(xml);
+
+            var resource = new Patient { Id = "test" };
+            var applier = new PatchApplicationService();
+            resource = (Patient)applier.Apply(resource, parameters);
+
+            Assert.Equal("1930-01-01", resource.BirthDate);
+        }
+
+        [Fact]
+        public async System.Threading.Tasks.Task WhenApplyingPropertyAssignmentPatchToNonEmptyPropertyThenThrows()
+        {
+            var xml = @"<Parameters xmlns=""http://hl7.org/fhir"">
+  <parameter>
+    <name value=""operation""/>
+    <part>
+      <name value=""type""/>
+      <valueCode value=""add""/>
+    </part>
+    <part>
+      <name value=""path""/>
+      <valueString value=""Patient""/>
+    </part>
+    <part>
+      <name value=""name""/>
+      <valueString value=""birthDate""/>
+    </part>
+    <part>
+      <name value=""value""/>
+      <valueDate value=""1930-01-01""/>
+    </part>
+  </parameter>
+</Parameters>";
+            var parser = new FhirXmlParser();
+            var parameters = parser.Parse<Parameters>(xml);
+
+            var resource = new Patient {Id = "test", BirthDate = "1930-01-01"};
+            var applier = new PatchApplicationService();
+
+            await Assert
+                .ThrowsAsync<TargetInvocationException>(() => Task.Run(() => applier.Apply(resource, parameters)))
+                .ConfigureAwait(false);
```

Comment:
Use Throws instead of ThrowsAsync and change the Test method to be synchronous

Reply:
Changed.